File: apps/catalog/app/schemas/goals.py
from datetime import date as date_type
from typing import Optional
from uuid import UUID

from pydantic import BaseModel


class GoalCreate(BaseModel):
    title: str
    description: Optional[str] = None
    target_value: float
    current_value: float
    unit: str
    category: str
    deadline: Optional[date_type] = None


class GoalUpdate(BaseModel):
    current_value: Optional[float] = None
    title: Optional[str] = None
    description: Optional[str] = None
    target_value: Optional[float] = None
    deadline: Optional[date_type] = None


class GoalResponse(BaseModel):
    id: UUID
    user_id: UUID
    title: str
    description: Optional[str]
    target_value: float
    current_value: float
    unit: str
    category: str
    deadline: Optional[date_type]
    created_at: str

    class Config:
        from_attributes = True

File: apps/backend/src/mastra/api/goals.ts
import { api, defaultConfig } from "../clients/catalog-client";
import type { ClientConfig, NutriaResponse } from "../clients/catalog-client";

export interface AddGoalRequest {
  title: string;
  description?: string;
  target_value: number;
  current_value: number;
  unit: string;
  category: string;
  deadline?: string;
}

export interface GoalResponse {
  id: string;
  title: string;
  target_value: number;
  current_value: number;
  unit: string;
}

export const addGoal = (
  request: AddGoalRequest,
  config: ClientConfig = defaultConfig,
  authToken?: string,
): Promise<NutriaResponse<GoalResponse>> =>
  api.post<GoalResponse>("/api/v1/goals", request, config, authToken);
